src/main/java/sirius/biz/tycho/kb/KnowledgeBaseSearchProvider.java

```diff
@@ -48,6 +48,10 @@ public boolean ensureAccess() {
     @Override
     public void query(String query, int maxResults, Consumer<OpenSearchResult> resultCollector) {
         knowledgeBase.query(UserContext.getCurrentUser().getLanguage(), query, maxResults).forEach(article -> {
+            KnowledgeBaseArticle manual = article.queryParent().orElse(null);
+            while (!KnowledgeBase.ROOT_CHAPTER_ID.equals(manual.queryParent().orElse(null).getArticleId())) {
```

Comment:
**Required:** `manual.queryParent().orElse(null).getArticleId()` this can lead to a NPE, or not? Better to use `Optional::map` when `queryParent` returns an `Optional`.